code/_onclick/hud/screen_objects/text_objects.dm

```diff
@@ -8,3 +8,123 @@
 	screen_loc = "CENTER-7,CENTER-7"
 	maptext_height = 480
 	maptext_width = 480
+
+/// A screen object that shows the time left on a timer
+/atom/movable/screen/text/screen_timer
+	screen_loc = "CENTER-7,CENTER-7"
+	var/maptext_style_left = "<span class='maptext' style=font-size:16pt;text-align:center; align='top'>"
+	var/maptext_style_right = "</span>"
+	var/maptext_string
+	var/timer_id
+	var/list/timer_mobs = list()
+
+/atom/movable/screen/text/screen_timer/Initialize(
+		mapload,
+		mobs,
+		timer,
+		text,
+		style_start,
+		style_end,
+		offset_x = 12,
+		offset_y = -70
+	)
+	. = ..()
+
+	if(!islist(mobs))
+		if(!mobs)
```

Comment:
you should probably .Copy the list if its passed a list in case someone passes a list that they're using